comms/AttributeType.cpp

```diff
@@ -375,6 +375,36 @@ bool IsValidExtension(const std::_tstring & ext)
         return true;
     return false;
 }
+bool IsValidExtensionDotted(const std::_tstring & ext)
+{
+    if (boost::algorithm::iends_with(ext, std::_tstring(_T(".")) + ATTRIBUTE_TYPE_ECL))
+        return true;
+    else if (boost::algorithm::iends_with(ext, std::_tstring(_T(".")) + ATTRIBUTE_TYPE_ECLMOD))
+        return true;
+    else if (boost::algorithm::iends_with(ext, std::_tstring(_T(".")) + ATTRIBUTE_TYPE_ECLLIB))
+        return true;
+    else if (boost::algorithm::iends_with(ext, std::_tstring(_T(".")) + ATTRIBUTE_TYPE_ESDL))
+        return true;
+    else if (boost::algorithm::iends_with(ext, std::_tstring(_T(".")) + ATTRIBUTE_TYPE_ECM))
+        return true;
+    else if (boost::algorithm::iends_with(ext, std::_tstring(_T(".")) + ATTRIBUTE_TYPE_SALT))
+        return true;
+    else if (boost::algorithm::iends_with(ext, std::_tstring(_T(".")) + ATTRIBUTE_TYPE_KEL))
+        return true;
+    else if (boost::algorithm::iends_with(ext, std::_tstring(_T(".")) + ATTRIBUTE_TYPE_DUD))
+        return true;
+    else if (boost::algorithm::iends_with(ext, std::_tstring(_T(".")) + ATTRIBUTE_TYPE_CMP))
+        return true;
+    else if (boost::algorithm::iends_with(ext, std::_tstring(_T(".")) + ATTRIBUTE_TYPE_PLUGIN))
+        return true;
+    else if (WildMatch(ext, std::_tstring(_T(".")) + ATTRIBUTE_TYPE_ECLOTHER, true))
+        return true;
+    else if (boost::algorithm::iends_with(ext, std::_tstring(_T(".")) + ATTRIBUTE_TYPE_DEFT))
+        return true;
+    else if (boost::algorithm::iends_with(ext, std::_tstring(_T(".")) + ATTRIBUTE_TYPE_LUCIFAMILY))
+        return true;
+    return false;
+}
```

Comment:
I think we can avoid having _another_ big switch statement here?

If this was renamed "filePathHasValidExtension" and the code extracted the extension (without the ".") it could simply call IsValidExtension?